yt_dlp/extractor/instagram.py

```diff
@@ -356,112 +358,153 @@ def _extract_embed_url(webpage):
         if mobj:
             return mobj.group('link')
 
+    def _get_pk(self, shortcode):
+        """Covert a shortcode to a numeric value."""
+        shortcode = shortcode[:11]
+        base = len(_ENCODING_CHARS)
+        strlen = len(shortcode)
+        num = 0
+        idx = 0
+        for char in shortcode:
+            power = strlen - (idx + 1)
+            num += _ENCODING_CHARS.index(char) * (base ** power)
+            idx += 1
+        return num
+
     def _real_extract(self, url):
         video_id, url = self._match_valid_url(url).group('id', 'url')
-        webpage, urlh = self._download_webpage_handle(url, video_id)
-        if 'www.instagram.com/accounts/login' in urlh.geturl():
-            self.report_warning('Main webpage is locked behind the login page. '
-                                'Retrying with embed webpage (Note that some metadata might be missing)')
-            webpage = self._download_webpage(
-                'https://www.instagram.com/p/%s/embed/' % video_id, video_id, note='Downloading embed webpage')
-
-        shared_data = self._parse_json(
-            self._search_regex(
-                r'window\._sharedData\s*=\s*({.+?});',
-                webpage, 'shared data', default='{}'),
-            video_id, fatal=False)
-        media = traverse_obj(
-            shared_data,
-            ('entry_data', 'PostPage', 0, 'graphql', 'shortcode_media'),
-            ('entry_data', 'PostPage', 0, 'media'),
-            expected_type=dict)
-
-        # _sharedData.entry_data.PostPage is empty when authenticated (see
-        # https://github.com/ytdl-org/youtube-dl/pull/22880)
-        if not media:
-            additional_data = self._parse_json(
+        webpageh = self._download_webpage_handle(url, video_id, fatal=False)
+        if not webpageh:
+            self.raise_no_formats('Requested content was not found')
+        else:
+            webpage, urlh = webpageh
```

Comment:
Why not just fatal=True?

Reply:
So can we do the thing with errnote?

Reply:
yeah, that works

Reply:
i checked it but still gives a 404 error message

Reply:
done